Runtime/AvatarCreator/UI/Elements/SelectionElement.cs

```diff
@@ -0,0 +1,69 @@
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace ReadyPlayerMe.AvatarCreator
+{
+    public interface IAssetData
+    {
+        public string Id { get; set; }
+        public AssetType AssetType { get; set; }
+    }
+
+    public abstract class SelectionElement : MonoBehaviour
+    {
+        [Header("UI Elements")]
+        [Space(5)]
+        [SerializeField] private ButtonElement buttonElementPrefab;
+        [SerializeField] private Transform buttonContainer;
+        [SerializeField] private GameObject selectedIcon;
+        public UnityEvent<IAssetData> onAssetSelected;
+        private Dictionary<string, ButtonElement> buttonMap = new Dictionary<string, ButtonElement>();
```

Comment:
I was using map initially, but I don't like it. Nowadays I prefer something like buttonElementById which sound much more descriptive.

Reply:
updated